openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java

```diff
@@ -2121,18 +2121,30 @@ private String computeDefaultSchemaName(Element definingElement, Element type, M
         if (metaAnnName != null && !io.swagger.v3.oas.annotations.media.Schema.class.getName().equals(metaAnnName)) {
             return NameUtils.getSimpleName(metaAnnName);
         }
+        String fullClassName;
         String javaName;
         if (type instanceof TypedElement) {
             ClassElement typeType = ((TypedElement) type).getType();
+            fullClassName = typeType.getName();
             if (CollectionUtils.isNotEmpty(typeType.getTypeArguments())) {
                 javaName = computeNameWithGenerics(typeType, typeArgs, context);
             } else {
                 javaName = computeNameWithGenerics(typeType, Collections.emptyMap(), context);
             }
         } else {
             javaName = type.getSimpleName();
+            fullClassName = type.getName();
         }
-        return javaName.replace("$", ".");
+        String packageName = fullClassName.substring(0, fullClassName.lastIndexOf('.'));
+        OpenApiApplicationVisitor.SchemaDecorator schemaDecorator = OpenApiApplicationVisitor.getSchemaDecoration(packageName, context);
+        javaName = javaName.replace("$", ".");
```

Comment:
you should use `getCanonicalName()` on the element instead of this hack

Reply:
What do you mean? If you are talking about substring by dollar symbol, then this is not my code, it was written that way. And I didn't understand how you suggest to use canoncialName ? getCanoncialName returns the full name of the class along with the package, but I don't need it, I need either the class name or the package name.

In any case, I changed the logic for calculating the package name.